frontend/javascripts/dashboard/dataset/queries.tsx

```diff
@@ -490,20 +485,22 @@ function diffDatasets(
     onlyB: onlyInNew,
     both,
   } = Utils.diffArrays(
-    oldDatasets.map((ds) => ds.name),
-    newDatasets.map((ds) => ds.name),
+    oldDatasets.map((ds) => ds.id),
+    newDatasets.map((ds) => ds.id),
   );
 
-  const oldDatasetsDict = _.keyBy(oldDatasets, (ds) => ds.name);
-  const newDatasetsDict = _.keyBy(newDatasets, (ds) => ds.name);
+  const oldDatasetsDict = _.keyBy(oldDatasets, (ds) => ds.id);
+  const newDatasetsDict = _.keyBy(newDatasets, (ds) => ds.id);
 
   const changedDatasets = both
-    .map((name) => newDatasetsDict[name])
+    .map((id) => newDatasetsDict[id])
     .filter((newDataset) => {
-      const oldDataset = oldDatasetsDict[newDataset.name];
-      return !_.isEqualWith(oldDataset, newDataset, (_objValue, _otherValue, key) => {
-        if (key === "lastUsedByUser") {
-          // Ignore the lastUsedByUser timestamp when diffing datasets.
+      const oldDataset = oldDatasetsDict[newDataset.id];
+      return !_.isEqualWith(oldDataset, newDataset, (oldValue, newValue, key) => {
+        const didUpgradeToRenamableDS =
+          !("directoryName" in oldValue) && "directoryName" in newValue; // TODO: Can be remove after a few weeks / months.
+        if (key === "lastUsedByUser" || didUpgradeToRenamableDS) {
+          // Ignore the lastUsedByUser timestamp when diffing datasets and migrating datasets to new renamable version.
           return true;
```

Comment:
> → I got a “there are 64 updated datasets” toast in the dashboard afterwards. Maybe suppress that if that is easy. If not, it’s also ok as is.

I could not reproduce this toast :see_no_evil: (maybe I am blind).
But I tried to remove the toast anyway by hiding my dataset

@daniel-wer I included a todo to remove this check after some time, when most users should have at least once opened the dashboard and got the dataset in the updated version. What do you think about this "fix" by ignoring an updated dataset based on whether it was migrated to the new data schema?

=> This might hide datasets that have updated since the last visit, as they also got the `directoryName` changed. Do you think, that's ok?

Reply:
> => This might hide datasets that have updated since the last visit, as they also got the directoryName changed. Do you think, that's ok?

I think that's totally fair. This is more of a convenience feature.